src/main/fc/fc_msp.c

```diff
@@ -1635,15 +1667,45 @@ static mspResult_e mspFcProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 
 #ifdef USE_RTC6705
     case MSP_SET_VTX_CONFIG:
-        ;
-        uint16_t tmp = sbufReadU16(src);
-        if  (tmp < 40)
-            masterConfig.vtx_channel = tmp;
-        if (current_vtx_channel != masterConfig.vtx_channel) {
-            current_vtx_channel = masterConfig.vtx_channel;
-            rtc6705_soft_spi_set_channel(vtx_freq[current_vtx_channel]);
+        if (dataSize == 2) {
+            uint16_t tmp = sbufReadU16(src);
+            if  (tmp < 40)
+                masterConfig.vtx_channel = tmp;
+            if (current_vtx_channel != masterConfig.vtx_channel) {
+                current_vtx_channel = masterConfig.vtx_channel;
+                rtc6705_soft_spi_set_channel(vtx_freq[current_vtx_channel]);
+            }
+        }
+        else {
+            return MSP_RESULT_ERROR;
         }
         break;
+#elif defined(VTX_COMMON)
```

Comment:
Will this work, considering the client does not know if the firmware was built with `USE_RTC6705` or `VTX_COMMON` defined? I'd probably rather go for a format where the data for the `USE_RTC6705` case is at the beginning of the message (and empty and ignored if `USE_RTC6705` is not used), and the `VTX_COMMON` data is appended to it if `VTX_COMMON` is used.